packages/fx-core/src/plugins/solution/fx-solution/v2/executeUserTask.ts

```diff
@@ -704,3 +708,133 @@ export type ParamForRegisterTeamsAppAndAad = {
   endpoint: string;
   "root-path": string;
 };
+
+// TODO: handle VS scenario
+export function canAddSso(
+  solutionSettings: AzureSolutionSettings,
+  telemetryReporter: TelemetryReporter
+): Result<Void, FxError> {
+  // Can not add sso if feature flag is not enabled
+  if (!isAadManifestEnabled()) {
+    const e = new SystemError(
+      SolutionError.NeedEnableFeatureFlag,
+      getLocalizedString("core.addSso.needEnableFeatureFlag"),
+      SolutionSource
+    );
+    return err(e);
+  }
+
+  if (!(solutionSettings.hostType === HostTypeOptionAzure.id)) {
+    const e = new UserError(
+      SolutionError.AddSsoNotSupported,
+      getLocalizedString("core.addSso.onlySupportAzure"),
+      SolutionSource
+    );
+    return err(
+      sendErrorTelemetryThenReturnError(SolutionTelemetryEvent.AddResource, e, telemetryReporter)
+    );
+  }
+
+  // Check whether SSO is enabled
+  const activeResourcePlugins = solutionSettings.activeResourcePlugins;
+  const containSsoItem = solutionSettings.capabilities.includes(SsoItem.id);
+  const containAadPlugin = activeResourcePlugins.includes(PluginNames.AAD);
+  if (containSsoItem && containAadPlugin) {
+    // Throw error if sso is already enabled
+    const e = new UserError(
+      SolutionError.SsoEnabled,
+      getLocalizedString("core.addSso.ssoEnabled"),
+      SolutionSource
+    );
+    return err(
+      sendErrorTelemetryThenReturnError(SolutionTelemetryEvent.AddSso, e, telemetryReporter)
+    );
+  } else if (containSsoItem || containAadPlugin) {
+    // Throw error if the project is invalid
+    const e = new UserError(
+      SolutionError.InvalidSsoProject,
+      getLocalizedString("core.addSso.invalidSsoProject"),
+      SolutionSource
+    );
+    return err(
+      sendErrorTelemetryThenReturnError(SolutionTelemetryEvent.AddSso, e, telemetryReporter)
+    );
+  }
+
+  return ok(Void);
+}
+
+export async function addSso(
+  ctx: v2.Context,
+  inputs: Inputs,
+  localSettings: Json
+): Promise<Result<unknown, FxError>> {
+  ctx.telemetryReporter.sendTelemetryEvent(SolutionTelemetryEvent.AddSsoStart, {
+    [SolutionTelemetryProperty.Component]: SolutionTelemetryComponentName,
+  });
+
+  let solutionSettings = getAzureSolutionSettings(ctx);
+  if (!solutionSettings) {
+    // pure existing app
+    solutionSettings = {
+      name: TeamsAppSolutionNameV2,
+      version: "1.0.0",
+      hostType: "Azure",
+      capabilities: [],
+      azureResources: [],
+      activeResourcePlugins: [],
+    };
+    ctx.projectSetting.solutionSettings = solutionSettings;
+    //aad need this file
+    await fs.writeJSON(`${inputs.projectPath}/permissions.json`, DEFAULT_PERMISSION_REQUEST, {
```

Comment:
This file should not need when there is a aad.template.json file

Reply:
Good catch. Will update this, thx！